bitnami/nginx/templates/deployment.yaml

```diff
@@ -35,6 +35,23 @@ spec:
       {{- include "nginx.imagePullSecrets" . | nindent 6 }}
       {{- if .Values.affinity }}
       affinity: {{- include "common.tplvalues.render" (dict "value" .Values.affinity "context" $) | nindent 8 }}
+      {{- else }}
+      affinity:
+        {{- if eq .Values.podAffinityPreset "soft" }}
+        podAffinity: {{- include "common.affinity.pod.soft" (dict "component" "nginx" "context" $) | nindent 10 }}
+        {{- else if eq .Values.podAffinityPreset "hard" }}
+        podAffinity: {{- include "common.affinity.pod.hard" (dict "component" "nginx" "context" $) | nindent 10 }}
+        {{- end }}
+        {{- if eq .Values.podAntiAffinityPreset "soft" }}
```

Comment:
Couldn't we have an extra helper like:

```
{{- define "common.affinities.pods" -}}
  {{- if eq .type "soft" }}
    {{- include "common.affinities.pods.soft" . -}}
  {{- else if eq .type "hard" }}
    {{- include "common.affinities.pods.hard" . -}}
  {{- end -}}
{{- end -}}
```

And then use it like

```
podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.podAntiAffinityPreset "component" "nginx" "context" $) | nindent 10 }} 
```

We would avoid a lot of `if/else` in all the charts, don't you think?